router/server.go

```diff
@@ -102,23 +133,25 @@ func NewRouter(ctx context.Context, h models.HandlerInterface, port int) *Router
 		}
 		h.TokenHandler(w, req, provider, false)
 	})))
-	mux.Handle("/api/gettoken", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(
+	gMux.Handle("/api/gettoken", h.ProviderMiddleware(h.AuthMiddleware(h.SessionInjectorMiddleware(
 		func(w http.ResponseWriter, req *http.Request, _ *models.Preference, _ *models.User, provider models.Provider) {
 			provider.ExtractToken(w, req)
 		}))))
 
 	// TODO: have to change this too
-	mux.Handle("/favicon.ico", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	gMux.Handle("/favicon.ico", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.Header().Set("Cache-Control", "public, max-age=3600") // 1 hr
-		http.ServeFile(w, r, "../ui/out/static/img/meshery-logo.png")
+		http.ServeFile(w, r, "../ui/out/static/img/meshery-logo/meshery-logo.svg")
 	}))
 
-	mux.Handle("/", h.ProviderMiddleware(h.AuthMiddleware(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		handlers.ServeUI(w, r, "", "../ui/out/")
-	}))))
+	// gMux.Handle("/", h.ProviderMiddleware(h.AuthMiddleware(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+	// 	handlers.ServeUI(w, r, "", "../ui/out/")
+	// }))))
+
+	gMux.Methods("GET").PathPrefix("/").Handler(h.ProviderMiddleware(h.AuthMiddleware(http.StripPrefix("/", http.FileServer(http.Dir("../ui/out/"))))))
```

Comment:
We would still want to use the UI handler to serve our SPA, use the handler SPA method